src/components/fields/DateField.js

```diff
@@ -42,40 +40,124 @@ class DateField extends AskField {
     )
   }
 
-  validateAndSave (options) {
-    this.validate()
-    this.update(options)
+  validateAndSave(options) {
+    if (this.validate()) {
+      this.setState({ value: this.buildValue() });
+    } else {
+      this.setState({ value: '--' });
+    }
+    this.update(options);
   }
 
   // Interface methods
 
-  validate () {
-    let isValid = true
-    let isCompleted = false
-    isCompleted = !!this.state.value.length
-    this.setState({ isValid: isValid, completed: isCompleted })
-    return this.props.wrapper.required ? isValid && isCompleted : isValid
+  validate() {
+    let isValid = true, isCompleted = false;
+    isValid = this.dateIsValid();
+    isCompleted = !!this.state.value.length;
+    this.setState({ isValid: isValid, completed: isCompleted });
+    return !!this.props.wrapper.required ? isValid && isCompleted : isValid;
+  }
+
+  buildValue() {
+    return this.state.month + "-" + this.state.day + "-" + this.state.year;
+  }
+
+  getValue() {
+    return { value: this.buildValue() };
   }
 
-  getValue () {
-    return { value: +this.state.value }
+  getStyles() {
+    return Object.assign({},
+      styles.textInput,
+      this.props.isValid ? styles.valid : styles.error,
+      this.state.focused ? styles.focused : {},
+      { backgroundColor: this.props.theme.inputBackground }
+    );
   }
 
-  render () {
+  daysInMonth(month, year) {
+    // Using 1-based months with 0
+    return new Date(year, month, 0).getDate();
```

Comment:
The 0th day of a month in JS is the last day of the previous month. Maybe you meant to subtract 1 from the month (months are 0-indexed in JS)?